sycl/doc/extensions/DeviceGlobal/SYCL_INTEL_device_global.asciidoc

```diff
@@ -0,0 +1,795 @@
+  size_t numBytes = sizeof(T), size_t offset = 0);
+----
+|`T` must be device copyable.
+
+Copies _count_ bytes from the `device_global` _src_ to the pointer _dest_, starting at _offset_ bytes. _dest_ may be either a host or USM pointer.
+|====
+--
+
+
+== Non-normative: Future anticipated changes, not enabled within this version of the specification
+
+In a future version of this extension, it is expected that when {cpp}20 support is available and enabled, the `consteval` keyword will be used to enable compile-time constant initialization of the device allocations backing `device_global`.  This will simplify some coding patterns, compared with the current zero-initialization requirement.
+
+A sketch of the anticipated constructor interface is:
+
+[source,c++]
+----
+namespace sycl::ext::oneapi {
+
+template <typename T, typename PropertyListT = property_list<>>
+class device_global {
+public:
+  using element_type = std::remove_extent_t<T>; 
+  
+  static_assert(std::is_trivially_destructible_v<T>,
+      "Type T must be trivially destructible.");
+
+  #ifdef __cpp_consteval
+    device_global();
+
+    // device_global initializes underlying T with the args argument
+    template <typename... Args>
+    consteval explicit device_global(Args&&... args);
+  #else
+    static_assert(std::is_trivially_default_constructible_v<T>,
+                  "Type T must be trivially default constructable (until C++20 "
+                  "consteval is supported and enabled)");
+
+    device_global();
+  #endif // __cpp_consteval
+
+----
+
+The example below creates two global namespace scope `device_global` objects named `dm1` and `dm2`. `dm1` is default constructed with external linkage, while `dm2` is initialized and has internal linkage.
+
+[source,c++]
+----
+using namespace sycl;
+using namespace sycl::ext::oneapi;
+
+device_global<MyClass> dm1;
+static device_global<int[4]> dm2{1, 3, 5, 7};  // Requires C++20 to be enabled
+
+int main () {
+  sycl::queue Q;
+  Q.submit([&](sycl::handler& h) {
+    h.single_task([=]() {
+      int x = 5;
+      if (dm1.get().flag)
+        x = dm2[0];
+    });
+  });
+}
+----
+
+== Non-normative: Implementation hints
+
+`device_global` prioritizes usability over simplicity of implementation, and therefore adds requirements such as (1) that contents and addresses of the allocation on each device remain stable across changes to specialization constant values, and (2) that the allocation be accessible across `device_image` on the same device.  These requirements mean that the semantics of `device_global` do not match the semantics of SPIR-V module scope variables, and therefore may not be implementable exclusively using the SPIR-V feature in existing SPIR-V consuming implementations.
+
+Also note that there are no restrictions on passing (and subsequent dereferencing) of pointers obtained on a device from a `device_global`, between kernels on a device, including through storage to memory.
```

Comment:
"on the same device", is more explicit and clearer

Reply:
Fixed